src/fabric/src/fabric2_fdb.erl

```diff
@@ -1172,6 +1261,45 @@ check_db_version(#{} = Db, CheckDbVersion) ->
     end.
 
 
+soft_delete_db(Db) ->
+    #{
+        name := DbName,
+        tx := Tx,
+        layer_prefix := LayerPrefix,
+        db_prefix := DbPrefix
+    } = ensure_current(Db),
+
+    DbKey = erlfdb_tuple:pack({?ALL_DBS, DbName}, LayerPrefix),
+    Timestamp = list_to_binary(fabric2_util:iso8601_timestamp()),
+    DeletedDbKeyTuple = {?DELETED_DBS, DbName, Timestamp},
+    DeletedDbKey = erlfdb_tuple:pack(DeletedDbKeyTuple, LayerPrefix),
+    case erlfdb:wait(erlfdb:get(Tx, DeletedDbKey)) of
+        not_found ->
+            erlfdb:set(Tx, DeletedDbKey, DbPrefix),
+            erlfdb:clear(Tx, DbKey),
+            bump_db_version(Db);
+        _Val ->
+            erlang:error({deleted_database_exists, DbName})
```

Comment:
This error message isn't quite right since we can have multiple deleted databases, just not at the same timestamp. How about `{deletion_frequency_exceeded, DbName}`?